low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +36,97 @@ class ServerMap(ResourceMap):
 
     def __init__(self):
         """Initialize server"""
-        pass
+        self.server_map = {
+            "memory": self.get_mem_info
+            }
 
     def get_health_info(self, rpath):
         """
         Get health information of fru in given resource id
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        info = {}
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            info["last_updated"] = int(time.time())
+            for fru in self.server_map:
+                info.update({fru: self.server_map[fru]()})
+        else:
+            fru = None
+            fru_found = False
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_map.get(fru):
+                    fru_found = True
+                    info = self.server_map[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                f"Health provider doesn't have support for'{rpath}'.")
+        return info
 
+    def get_mem_info(self):
+        """Update and return memory info in specific format"""
 
-# if __name__ == "__main__":
-#     server = ServerMap()
-#     health_data = server.get_health_info(rpath="nodes[0]>compute[0]>hw>disks")
-#     print(health_data)
+        from framework.utils.conf_utils import SSPL_CONF, Conf
+        default_mem_usage_threshold = int(Conf.get(SSPL_CONF,
+            "NODEDATAMSGHANDLER>host_memory_usage_threshold",
+            80))
+        data = []
+        status = "OK"
+        description = "Host memory is in good health"
+        self.mem_info = dict(psutil.virtual_memory()._asdict())
+        curr_mem_usage_threshold = int(self.mem_info['percent'])
+        if curr_mem_usage_threshold > int(default_mem_usage_threshold):
+            status = "HIGH MEMORY USAGE"
+            description = (f"Host memory usage increased to {curr_mem_usage_threshold} "
+                f"beyond configured threshold of {default_mem_usage_threshold}")
+
+        memory_dict = self.prepare_mem_json(status,
+                                            description)
+        data.append(memory_dict)
+        return data
+
+    def prepare_mem_json(self, status, description):
+        """Update and return memory information dict """
+        total_memory = {}
+        for key, value in self.mem_info.items():
+            if key == 'percent':
+                total_memory['percent'] = str(self.mem_info['percent']) + '%'
+            else:
+                total_memory[key] = str(self.mem_info[key] >> 20) + 'MB'
+
+        memory_dict = {
+            "uid": "memory_0",
```

Comment:
lets keep 'main_memory' as it doesnt have any instance associated like 0,1, etc

Reply:
done